src/command/dev/next/mod.rs

```diff
@@ -63,26 +59,38 @@ impl Dev {
             .await
             .map_err(|err| RoverError::new(anyhow!("{}", err)))?;
 
-        let supergraph_yaml = self.opts.supergraph_opts.clone().supergraph_config_path;
-        let federation_version = self.opts.supergraph_opts.federation_version.clone();
-        let profile = self.opts.plugin_opts.profile.clone();
-        let graph_ref = self.opts.supergraph_opts.graph_ref.clone();
-
-        let one_shot_composition = OneShotComposition::builder()
-            .client_config(client_config.clone())
-            .profile(profile.clone())
-            .elv2_license_accepter(elv2_license_accepter)
-            .skip_update(skip_update)
-            .and_federation_version(federation_version)
-            .and_graph_ref(graph_ref.clone())
-            .and_supergraph_yaml(supergraph_yaml)
-            .and_override_install_path(override_install_path.clone())
-            .build();
-
-        let supergraph_schema = one_shot_composition
-            .compose(&read_file_impl, &write_file_impl, &exec_command_impl)
+        let profile = &self.opts.plugin_opts.profile;
+        let graph_ref = &self.opts.supergraph_opts.graph_ref;
+        let supergraph_config_path = &self.opts.supergraph_opts.clone().supergraph_config_path;
+
+        let service = client_config.get_authenticated_client(profile)?.service()?;
+        let service = WhoAmI::new(service);
+
+        let composition_pipeline = CompositionPipeline::default()
+            .init(
+                &mut stdin(),
+                &client_config.get_authenticated_client(profile)?,
```

Comment:
I really like this idea, because I actually had to implement something similar for the LSP (because they wanted composition to occur as soon as the pipeline starts). That being said, another requirement from the LSP work is that if you don't have anything in the `supergraph.yaml` that requires GraphOS you shouldn't need a profile to be configured. Because get authenticated client here causes a profile look up that won't work. Is there any way we can defer this to the subgraph level?

Reply:
Or could we have a new implementation of FetchRemoteSubgraphs that only initialises a concrete client at the subgraph level?

Reply:
I think it might be something where we can defer this until we scan the supergraph config for studio subgraphs. That way we can be up front about requiring the profile before we attempt to resolve the subgraphs. Would that work?

Reply:
If you could then great